Address-Book.go

```diff
@@ -200,21 +173,69 @@ func Check(w http.ResponseWriter, r *http.Request) {
 	}
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
-func HomePage(w http.ResponseWriter, r *http.Request) {
+func (MyApp *App) HomePage(w http.ResponseWriter, r *http.Request) {
 	fmt.Println("home")
-	templates := template.Must(template.ParseFiles("index.html"))
-	if err := templates.ExecuteTemplate(w, "index.html", nil); err != nil {
+	if err := MyApp.templates.ExecuteTemplate(w, "index.html", nil); err != nil {
 		fmt.Println("error home")
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 
 }
 ///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (MyApp *App) UserPage(w http.ResponseWriter, r *http.Request){
+	MyApp.User.Contacts = []Contact{}
+	session , _ := store.Get(r,"CurrentSession")
+	UsernameIn :=session.Values["user"]
+	Username :=UsernameIn.(string)
+	if Username==""{
+		http.Redirect(w,r,"/home",http.StatusFound)
+		return
+	}
+	MyApp.User.UserName =Username
+	rows, err := MyApp.db.Query("select contactID,fname,lname,email,id,phonenumber from contact join`phonenumbers` on contact.contactID = phonenumbers.contact_id where userID= ( select id from users where username = ?)" , Username)
```

Comment:
It is better to use join then to use inner select. If you need the userId a lot why not to put it in the session?